header/p2p/server.go

```diff
@@ -151,43 +159,85 @@ func (serv *ExchangeServer) requestHandler(stream network.Stream) {
 // if it exists.
 func (serv *ExchangeServer) handleRequestByHash(hash []byte) ([]*header.ExtendedHeader, error) {
 	log.Debugw("server: handling header request", "hash", tmbytes.HexBytes(hash).String())
+	ctx, cancel := context.WithTimeout(serv.ctx, time.Second*5)
+	defer cancel()
+	ctx, span := serverTracer.Start(ctx, "ExchangeServer_handleRequestByHash", trace.WithAttributes(
+		attribute.String("hash", tmbytes.HexBytes(hash).String()),
+	))
+	defer span.End()
 
-	h, err := serv.getter.Get(serv.ctx, hash)
+	h, err := serv.getter.Get(ctx, hash)
 	if err != nil {
 		log.Errorw("server: getting header by hash", "hash", tmbytes.HexBytes(hash).String(), "err", err)
+		span.RecordError(err)
```

Comment:
We have to set the status. It's not set automatically, as per docs

Reply:
done